metacat-connector-polaris/src/main/java/com/netflix/metacat/connector/polaris/PolarisConnectorTableService.java

```diff
@@ -182,16 +188,57 @@ public TableInfo get(final ConnectorRequestContext requestContext, final Qualifi
      */
     @Override
     public void update(final ConnectorRequestContext requestContext, final TableInfo tableInfo) {
+        final QualifiedName name = tableInfo.getName();
         try {
-            // TODO: implement once data model supports this operation
+            final Map<String, String> newTableMetadata = tableInfo.getMetadata();
+            if (MapUtils.isEmpty(newTableMetadata)) {
+                final String message = String.format("No parameters defined for iceberg table %s", name);
+                log.warn(message);
+                throw new InvalidMetaException(name, message, null);
+            }
+            final String prevLoc = newTableMetadata.get(DirectSqlTable.PARAM_PREVIOUS_METADATA_LOCATION);
+            final String newLoc = newTableMetadata.get(DirectSqlTable.PARAM_METADATA_LOCATION);
+            if (StringUtils.isBlank(prevLoc) || StringUtils.isBlank(newLoc)) {
+                final String message = String.format(
+                    "Invalid metadata for %s. Provided previous %s or new %s location is empty.",
+                    name, prevLoc, newLoc);
+                log.error(message);
+                throw new InvalidMetaException(name, message, null);
+            }
+            // optimistically attempt to update metadata location
+            final boolean updated = polarisConnector.updateTableMetadataLocation(
```

Comment:
Are we not storing other table properties?

Reply:
Not currently, we might start storing previous location as a separate field though.

Reply:
We'll have a sync to see if some of the other current use cases can be moved to use the Iceberg map instead.